### ISSUE TITLE
Should ResponsiveImageType include `null`?

### ISSUE THREAD

**Issue by calirvine** [datocms/react-datocms#51]
I don't see any other issues for this, so maybe I'm just completely missing something 😅

When using the `Image` component of react-dato, I get a type error on ResponsiveImageType, as my Graphql data is correctly typed as a union of the field type | null | undefined (as GraphQL returns null, not undefined for a field that is requested but does not exist), but the data type of `Image` is as follows:

```
export declare type ResponsiveImageType = {
    /** The aspect ratio (width/height) of the image */
    aspectRatio: number;
    /** A base64-encoded thumbnail to offer during image loading */
    base64?: string;
    /** The height of the image */
    height?: number;
    /** The width of the image */
    width: number;
    /** The HTML5 `sizes` attribute for the image */
    sizes?: string;
    /** The fallback `src` attribute for the image */
    src?: string;
    /** The HTML5 `srcSet` attribute for the image */
    srcSet?: string;
    /** The HTML5 `srcSet` attribute for the image in WebP format, for browsers that support the format */
    webpSrcSet?: string;
    /** The background color for the image placeholder */
    bgColor?: string;
    /** Alternate text (`alt`) for the image */
    alt?: string;
    /** Title attribute (`title`) for the image */
    title?: string;
};
```

Looking at the source code, it doesn't look like a strict check for `undefined` is performed anywhere except for on the window object for `isSsr`, therefore I don't think adding null to the types here would break any behaviour. 

I'm happy to make a PR for this if you agree.

**Comment by stefanoverna** [comment]
sure, go ahead!

**Comment by calirvine** [comment]
Merged in #52
